### DISCUSSION TITLE
What exactly is `SetTargetFPS`?

### DISCUSSION THREAD

**Discussion by sakgoyal** [raysan5/raylib#4201] [General]
What does that function do exactly? 
is it more of a "maximum" allowable fps? 
or does it dynamically change the resolution/quality to maintain the target fps so it doesnt go above or below? 

also, what happens if I do not set a target fps?

**Comment by ghost** [comment]


**Reply by sakgoyal** [reply]
> EndDrawing()

I assume you mean here?
https://github.com/raysan5/raylib/blob/9e39788e077f1d35c5fe54600f2143423a80bb3d/src/rcore.c#L908


I was basically asking because I wanted to know if there way a way I can do some sort of "delta time" calculation instead of relying on the framerate to do physics calculations. I don't want to rely on the framerate of the monitor (vsync). 

or should I just set a really high target like 200 or something and then do delta time calculations with that as that would work even if the system can't keep up with 200?

**Comment by casperbear** [comment]
For some reason I cannot see what [asdqwe](https://github.com/asdqwe) told you, but I played with SetTargetFPS quite a lot. My advice: call SetTargetFPS(60) in the beginning. Otherwise 1) if you don't use FLAG_VSYNC_HINT you will see multiple screen tearing (on multiple levels on the screen) if your app is simple enough and your GPU is powerful enough 2) if you do use FLAG_VSYNC_HINT then you will most likely see actual FPS worse than 60 (assuming your monitor refresh rate is 60) plus some microstuttering (I kind of understood why it happens but not well enough to try to explain it, just test it for yourself). SetTargetFPS basically adds additional wait to cap FPS at whatever value you pass. But you can use frameTime for physics calculations regardless of calling SetTargetFPS.
SetTargetFPS will not increase FPS to whatever value you pass. Now, I'll assume that your question is not a basic one, like you don't understand the simple fact that nothing can make your GPU produce more frames than it's capable of, and you actually asked whether SetTargetFPS is some RayLib way to provide you with a steady stream of callbacks of small frameTime values for your app to do the simulation stuff. No, it's not that, it's only for actual GPU/monitor FPS cap.

**Reply by sakgoyal** [reply]
thank you. this makes more sense and aligns more to what I expected it to be doing. 

for context. yes I am aware of the basics of gpus. 

so essentially it's just a frame cap. got it. 
I would much rather have tearing than vsync. vsync sucks as a gamer
